src/app/code/community/Shopgate/Cloudapi/Helper/Request.php

```diff
@@ -104,6 +104,14 @@ public function cookieIsSet($key, $value = false)
         return !$value ? true : $data[$key] === $value;
     }
 
+    /**
+     * @return string | false
+     */
+    public function userAgentIsShopgateApp()
+    {
+        return strstr(Mage::helper('core/http')->getHttpUserAgent(), 'libshopgate');
```

Comment:
```suggestion
        return strstr(Mage::helper('core/http')->getHttpUserAgent(), 'libshopgate') !== false;
```
Just return an bool in any case.